apps/apps.humanatlas.io/src/app/pages/us1/cell-population-predictor/rui/embedded-rui.component.ts

```diff
@@ -0,0 +1,90 @@
+import { CommonModule, DOCUMENT } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  CUSTOM_ELEMENTS_SCHEMA,
+  effect,
+  ElementRef,
+  inject,
+  input,
+  output,
+  Renderer2,
+  viewChild,
+} from '@angular/core';
+
+/**
+ * RUI Interface
+ */
+interface RuiElement {
+  organOptions: string[];
+
+  register: (location: object) => void;
+  cancelRegistration: () => void;
+}
+
+/** RUI script URL */
+const SCRIPT_URL = 'https://cdn.humanatlas.io/ui/ccf-rui/wc.js';
+/** RUI style URL */
+const STYLE_URL = 'https://cdn.humanatlas.io/ui/ccf-rui/styles.css';
+
+/**
+ * Embedded Registration User Interface Component
+ */
+@Component({
+  selector: 'hra-embedded-rui',
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: './embedded-rui.component.html',
+  styleUrl: './embedded-rui.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  schemas: [CUSTOM_ELEMENTS_SCHEMA],
+})
+export class EmbeddedRuiComponent {
+  /** Supported Organs */
+  readonly supportedOrgans = input<string[]>([]);
+  /** File Created after Registration */
+  readonly locationCreated = output<File>();
+  /** Close RUI */
+  readonly closed = output();
+  /** Reference to the rui element in template */
+  private readonly ref = viewChild.required<ElementRef<RuiElement>>('rui');
+  /** For accessing DOM  */
+  private readonly document = inject(DOCUMENT);
+  /** For manuplating DOM elements */
+  private readonly renderer = inject(Renderer2);
+
+  constructor() {
+    this.setupScriptAndStyleTags();
+
+    effect(() => {
+      const el = this.ref().nativeElement;
+      el.organOptions = this.supportedOrgans();
+      el.register = (location) => this.locationCreated.emit(this.createFileFromLocation(location));
+      el.cancelRegistration = () => this.closed.emit();
+    });
+  }
+
+  private createFileFromLocation(location: object): File {
+    return new File([JSON.stringify(location)], 'rui-location.json');
+  }
+
+  private setupScriptAndStyleTags(): void {
```

Comment:
Let's move this code into `cell-population-predictor` so that the script and styles are loaded earlier and the rui can be opened more quickly